### ISSUE TITLE
Update hidden input field value while using FilePond fetch.

### ISSUE THREAD

**Issue by rahulhaque** [pqina/filepond#714]
**Describe the bug**

I am not sure if its a bug or what I am asking is a feature. The thing is filepond's `./process` URL uploads the file first and then append a hidden input field with the unique id returned from the server. When I submit the form, that hidden input value is also submitted based on which I can determine what to do with the temporary file.

However with the filepond `./fetch` URL, the end point is used to load files located on remote servers. It asks the server to download the file and return a file object. An alternative is to configure FilePond to only do a HEAD request. The server should then save the file and return response headers with the required file information. 

When I return response headers with the required information, filepond append a hidden input field. But the field does not contain any value returned from the server. Is this normal? If it is, is there any way to set the hidden field value?

Here is my filepond's default configuration.
```javascript
FilePond.setOptions({
  server: {
      process: '/filepond',
      revert: '/filepond',
      fetch: {
          url: '/filepond?data=',
          method: 'HEAD',
      },
      headers: {
          'X-CSRF-TOKEN': document.head.querySelector("[name=csrf-token][content]").content,
      }
  }
});
```

**Logs**

Here is the response header returned from the server for `./fetch` url.
```javascript
HTTP/1.1 200 OK
Server: nginx/1.18.0
Content-Type: image/png
Content-Length: 17496
Access-Control-Expose-Headers: Content-Disposition, Content-Length, X-Content-Transfer-Id
Content-Disposition: inline; filename="123456"
X-Content-Transfer-Id: 123456
```

**To Reproduce**

To reproduce, I didn't find any `./fetch` example online. I am tinkering by myself.

**Expected behavior**

I just want the hidden input value to be set by any of the header value when it receives response header from `./fetch` url just like the `./process` url does when it gets the unique id from the server. Pardon my lack of knowledge about filepond if this feature is already available.

**Information about my project:**

- Brave browser V1.24.86 Chromium: 90.0.4430.212 (Official Build) (64-bit)
- Elementary OS Hera 5.1
- FilePond version 4.27.1

**Additional context**

After successful `./fetch` url request, filepond appends the below hidden input field. The value you see here is the remote server address of the file. I just want this value to be unique id returned from my backend, in this case - 123456
```html
<input type="hidden" name="avatar" value="https://www.somesite.xyz/logo.png">
```

Btw thank you for this amazing library. I have made a package for Laravel that gracefully handles filepond's `./process` and `./revert` requests. I was thinking of implementing `./fetch` url process but the documentation didn't take me far.

**Comment by rikschennink** [comment]
We require the issue template to be filled out on all new issues. The issue template helps us collect all the information we need to address your submission efficiently.

If your issue can be classified as an integration question or a question about specific features of FilePond we ask you kindly to submit it on [Stack Overflow](https://stackoverflow.com/questions/tagged/filepond).

We will be able to more closely look at your issue once the issue template has been filled out. The issue will be closed until the above is addressed.

**Please leave a comment after you've updated the issue.**

**Comment by rahulhaque** [comment]
@rikschennink sorry for bringing you in this again. I hope you will understand and clear my confusion regarding usage of filepond's `./fetch` endpoint.

**Comment by rikschennink** [comment]
@rahulhaque Have you uploaded the file? So the remote file is added, then there's an upload action (either instant or manual) and then the unique id is set. Otherwise it's just treated as a normal file but from a remote location.

**Comment by rahulhaque** [comment]
@rikschennink Yes, I have uploaded the file. Here are the steps.

- Before dropping file.
![1](https://user-images.githubusercontent.com/12048213/119785163-c4e98980-bef0-11eb-8372-8438b9e2276b.png)

- After dragging and dropping the logo from https://laravelpackage.com/ it sends the logo url to my backend like this.
```javascript
Request URL: http://shodai-support.test/filepond?data=https%3A%2F%2Fd33wubrfki0l68.cloudfront.net%2F25c7fa957c5f79268d59b1516d12a0ea9effadaf%2F13bc8%2Flaravel-package-logo.png
Request Method: HEAD
Status Code: 200 OK
Remote Address: 127.0.0.1:80
Referrer Policy: strict-origin-when-cross-origin
```
I download the file in backend and return response with the header mentioned in my previous comment like this.
```javascript
Access-Control-Expose-Headers: Content-Disposition, Content-Length, X-Content-Transfer-Id
Cache-Control: no-cache, private
Connection: keep-alive
Content-Disposition: inline; filename="123456"
Content-Length: 8947
Content-Type: image/png
Date: Thu, 27 May 2021 07:59:15 GMT
Server: nginx/1.18.0
X-Content-Transfer-Id: 123456
```

Filepond shows success. 
![Screen record from 2021-05-27 13 59 21](https://user-images.githubusercontent.com/12048213/119788315-db451480-bef3-11eb-89f2-e25f56622dc4.gif)

- Filepond appends the hidden field in the form like this. You see the input value contains the logo url even after I returned unique id in the header.
```html
<input type="hidden" name="avatar" value="https://d33wubrfki0l68.cloudfront.net/25c7fa957c5f79268d59b1516d12a0ea9effadaf/13bc8/laravel-package-logo.png">
```

- **The interesting part** is if I click the close button, the `./revert` request is sent with the unique id.
![3](https://user-images.githubusercontent.com/12048213/119791244-78a14800-bef6-11eb-93eb-50c207302027.png)
```javascript
Request URL: http://shodai-support.test/filepond
Request Method: DELETE
Status Code: 200 OK
Remote Address: 127.0.0.1:80
Referrer Policy: strict-origin-when-cross-origin
----------------------------------------------------------------
Request Payload: 123456
```

**Comment by rikschennink** [comment]
Should be fixed in `4.27.2`, just published.

**Comment by rahulhaque** [comment]
Updated and tested. It's fixed now. Thanks a lot for your effort. Now I can happily implement the FilePond fetch endpoint in my package. 😀️ Have a nice day.
